inductiva/tasks/task.py

```diff
@@ -130,6 +132,12 @@ def wait(self, polling_period: int = 5) -> TaskStatusCode:
             prev_status = status
 
             if status in terminal_statuses:
+                if self.delete_machine:
+                    delete_logging_info = f"Deleting the machine \
+                                        {self.get_machine()}"
+
+                    logging.info(delete_logging_info)
+                    admin.kill_executer(self.get_machine())
```

Comment:
This will not work as you expect. You can only kill an executer if it does not belong to a group.

Moreover, this implementation only works for sync runs and I am not sure if that fulfils your purpose.